app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -122,9 +122,14 @@ def set_current_user(skip_terms_check)
     return unless @session_object
 
     user = User.find(@session_object.uuid)
+    validate_credential_lock(user:)
     @current_user = user if skip_terms_check || !user&.needs_accepted_terms_of_use
   end
 
+  def validate_credential_lock(user:)
+    raise Login::Errors::CredentialLockedError, 'Credential is locked' if user&.credential_lock
```

Comment:
I would just not set `@current_user = user` if `user.credential_lock == true`, and let the normal `unauthorized` error get raised in that case. Reason is that the frontend might get confused over a `:forbidden` status